test/functional/extractItem.test.ts

```diff
@@ -0,0 +1,202 @@
+// ----------------------------------------------------------------------------
```

Comment:
The following doesn't work (don't think my code broke it??):
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.2.2.3",
    "resources": [
        {
            "type": "Microsoft.Compute/virtualMachines/extensions",
            "apiVersion": "2019-12-01",
            "name": "[concat(parameters('vmName'),copyIndex(1),'/dscext')]",
            "dependsOn": [
                // >>> SET CURSOR ON FOLLOWING LINE BETWEEN "r" and "esourceId"
                "[resourceId('Microsoft.Compute/virtualMachines/extensions', concat(parameters('vmName'),copyIndex(1),'/dscext'))]"
            ]
        }
    ]
}
```
Extracting any parameter ends up with a syntax error:
```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.2.2.3",
    "resources": [
        {
            "type": "Microsoft.Compute/virtualMachines/extensions",
            "apiVersion": "[parameters('abc')]",
            "name": "[concat(parameters('vmName'),copyIndex(1),'/dscext')]",
            "dependsOn": [
                // >>> SET CURSOR ON FOLLOWING LINE BETWEEN "r" and "esourceId"
                "[resourceId('Microsoft.Compute/virtualMachines/extensions', concat(parameters('vmName'),copyIndex(1),'/dscext'))]"
            ],
            "parameters": {
                "abc": {
                    "type": "string",
                    "defaultValue": "2019-12-01"
                }
            }
        }
    ]
}
```

Reply:
Great find! This is fixed now.